packages/hardhat-core/test/builtin-tasks/flatten.ts

```diff
@@ -105,4 +129,213 @@ describe("Flatten task", () => {
       }
     });
   });
+
+  describe("SPDX licenses and pragma abicoder directives", () => {
+    describe("Flatten files that not contain SPDX licenses or pragma directives", () => {
+      useFixtureProject("flatten-task/contracts-no-spdx-no-pragma");
+
+      it("should successfully flatten and compile the files", async function () {
+        const [flattenedFiles, _metadata] = await this.env.run(
+          TASK_FLATTEN_GET_FLATTENED_SOURCE_AND_METADATA
+        );
+
+        await assertFlattenedFilesResult(flattenedFiles);
+      });
+    });
+
+    describe("Flatten files that contain SPDX licenses", () => {
+      describe("Files contain one single license per file", () => {
+        describe("Files contain same licenses", () => {
+          useFixtureProject("flatten-task/contracts-spdx-same-licenses");
+
+          it("should successfully flatten and compile the files", async function () {
+            const [flattenedFiles, _metadata] = await this.env.run(
```

Comment:
Should we be asserting the content of the returned metadata in these tests?

Reply:
done